sktime/forecasting/compose/_grouped.py

```diff
@@ -122,3 +131,554 @@ def get_test_params(cls, parameter_set="default"):
+
+    Parameters
+    ----------
+    forecasters : dict[sktime forecasters]
+        dict of forecasters with the key corresponding to categories generated
+        by the given transformer and the value corresponding to a sktime forecaster.
+
+    transformer : sktime transformer, default = ADICVTransformer()
+        A series-to-primitives sk-time transformer that generates a value
+        which can be used to quantify a choice of forecaster for the time series.
+
+        Note: To ensure correct functionality, the transformer must store the
+        generated category in the first column of the returned values when
+        the transform() or fit_transform() functions are called.
+
+    fallback_forecaster : sktime forecaster | None, Optional
+        A fallback forecaster that will be used if the category generated by
+        the transformer does not match any of the given forecasters.
+
+    Raises
+    ------
+    AssertionError: If a valid transformer (an instance of BaseTransformer)
+    is not passed or if valid forecasters (instances of BaseForecaster) are not given.
+
+    Examples
+    --------
+    This example showcases how the GroupbyCategoryForecaster can be utilized to select
+    appropriate forecasters on the basis of the time series category determined by
+    the ADICVTransformer!
+
+    >>> from sktime.forecasting.compose import GroupbyCategoryForecaster
+    >>> from sktime.forecasting.croston import Croston
+    >>> from sktime.forecasting.trend import PolynomialTrendForecaster
+    >>> from sktime.forecasting.naive import NaiveForecaster
+    >>> from sktime.transformations.series.adi_cv import ADICVTransformer
+
+    # Importing the methods which can generate data of specific categories
+    depending on their variance and average demand intervals.
+
+    >>> from sktime.transformations.series.tests.test_adi_cv import (
+    ...     _generate_erratic_series)
+
+    # The forecaster is defined which accepts a dictionary of forecasters,
+    a transformer and optionally a fallback_forecaster
+
+    >>> group_forecaster = GroupbyCategoryForecaster(
+    ...     forecasters =
+    ...         {"smooth": NaiveForecaster(),
+    ...         "erratic": Croston(),
+    ...         "intermittent": PolynomialTrendForecaster()},
+    ...     transformer=ADICVTransformer(features=["class"]))
+
+    >>> generated_data = _generate_erratic_series()
+
+    # The fit function firstly passes the data through the given transformer
+    # to generate a given category. This category can be seen by the variable
+    # self.category_.
+
+    >>> group_forecaster = group_forecaster.fit(generated_data, fh=50)
+    >>> #print(f"The chosen category is: {group_forecaster.category}")
+
+    >>> # Print out the predicted value over the given forecasting horizon!
+    >>> # print(group_forecaster.predict(fh=50, X=None))
+    """
+
+    _tags = {
+        "y_inner_mtype": [
+            "pd.DataFrame",
```

Comment:
I think the faiure is due to not handling the case of a single time series correctly.

If you want to deal only with MultiIndex, you can remove the non-multiindexed data types, and afaik the boilerplate will add a dummy level with a single value.

Reply:
oh perfect. Will do that

Reply:
oh, ouch, it turns out I misremembered - there is some code duplication between `BaseTransformer` and `BaseForecaster`, but the feature where a single series is coerced to a panel is only present in the duplicate location in `BaseTransformer`, which is ahead of the `BaseForecaster` that does not have this.

I would hence suggest:

* open an issue on the lack of boilerplate functionality in this csae
* add back `"pd.DataFrame"` to the internal types and deal with it somehow - worst case one could copy the code from the `TransformSelectForecaster`

Reply:
hmm I understand. I will handle this case in the GroupbyCategoryForecaster class then.